plugin/evm/orderbook/memory_database.go

```diff
@@ -239,7 +261,13 @@ func (db *InMemoryDatabase) Accept(blockNumber uint64) {
 		lifecycle := order.getOrderStatus()
 		if (lifecycle.Status == FulFilled || lifecycle.Status == Cancelled) && lifecycle.BlockNumber <= blockNumber {
 			delete(db.OrderMap, orderId)
+			continue
+		}
+		expireAt := order.getExpireAt()
+		if expireAt.Cmp(big.NewInt(0)) > 0 && expireAt.Int64() <= time.Now().Unix() {
```

Comment:
- use `.Sign()`, it reads better.
- See if you can use parent block's timestamp instead of `time.Now().Unix()`
- remove equality from the comparison. There is a slight chance that the next block has the same timestamp.